src/xdocs/checks/coding/missingnullcaseinswitch.xml

```diff
@@ -0,0 +1,142 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<document xmlns="http://maven.apache.org/XDOC/2.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/XDOC/2.0 https://maven.apache.org/xsd/xdoc-2.0.xsd">
+  <head>
+    <title>MissingNullCaseInSwitch</title>
+  </head>
+  <body>
+    <section name="MissingNullCaseInSwitch">
+      <p>Since Checkstyle 10.18.0</p>
+      <subsection name="Description" id="Description">
+        <p>
+          Checks that a given switch statement or expression that use a reference type
```

Comment:
I am slightly confused as the check is named `...InSwitch` not `...InSwitchOrExpression...`.

You are referring to switch expressions? Or is how you wrote it the way it is worded in the JLS? (Link?)

Reply:
Yes, I was referring to switch expressions. I didn't take this wording from JLS. I took it from existing documentation in our checks.

https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/c30e2a6_2024151038/checks/coding/missingswitchdefault.html

> See the [Java Language Specification](https://docs.oracle.com/javase/specs/jls/se22/html/jls-15.html#jls-15.28) for more information about switch statements and expressions. 

the word switch in the check name is a general name for any switch. I think we shouldn't name it `InSwitchOrExpression`. we don't have to be very explicit in the name. but in the description I added it to make it clear that we verify both as both are subjected to NPE if selector expression is null.

Reply:
This is pedantic IMO, but we can do ‘switch statement or switch expression’ here. The existing wording should be understood as ‘switch (statement or expression)’

Reply:
Ok, I think my mind just went a different direction when I was reading this.